### ISSUE TITLE
[BUG] UnusedCustomPropertiesRule false positive for parent object property when only nested children are referenced

### ISSUE THREAD

**Issue by alex-spyksma** [bw-design-group/ignition-lint#97]
## Describe the bug

`UnusedCustomPropertiesRule` falsely reports a **parent** custom property as "defined but never referenced" when that property is an **object/container** whose **nested child properties** are the things actually referenced and bound.

If a custom property is an object (e.g. `custom.network`) and its children (`custom.network.nat1`, `custom.network.nat2`) have bindings and are referenced in scripts/expressions, the parent `custom.network` is necessarily used — the children cannot exist without it. The rule only tracks the **top-level segment name** and never credits a parent for the usage of its descendants, so it emits a false positive on the parent.

## Root cause

In `src/ignition_lint/rules/properties/unused_custom_properties.py`:

1. **Definitions are top-level only.** `visit_property` only records a definition when there is no dot after the prefix (`'.' not in path[7:]`), so `custom.network` is tracked but `custom.network.nat1` / `custom.network.nat2` are not (lines 99–108).

2. **Child references don't credit the parent.**
   - The script scanner (`_check_script_for_references`, lines 242–260) matches `self\.custom\.([a-zA-Z_][a-zA-Z0-9_]*)`, which captures only `network` from `self.custom.network.nat1` and adds the **wildcard component-level** key `*.custom.network`. But the definition is the **view-level** key `view.custom.network`, and the wildcard fallback in `finalize` is explicitly skipped for view-level props (`not prop_path.startswith('view.')`, line 282). So `view.custom.network` never matches.
   - `_mark_binding_owner_as_used` is called for the child bindings (`custom.network.nat1`, `custom.network.nat2`), but it marks `view.custom.network.nat1` / `view.custom.network.nat2` (or doesn't match cleanly) — never the parent `view.custom.network`.

The core gap: there is no logic that says *"a defined property is used if any of its child paths are referenced or bound."*

## To Reproduce

Minimal sanitized `view.json` (proprietary content removed):

```json
{
  "custom": {
    "network": {}
  },
  "propConfig": {
    "custom.network": {
      "onChange": {
        "enabled": null,
        "script": "\tif self.custom.network.nat1 == \"0.0.0.0\" and self.custom.network.nat2 == \"0.0.0.0\":\n\t\tpass"
      },
      "persistent": true
    },
    "custom.network.nat1": {
      "binding": {
        "config": { "mode": "indirect", "tagPath": "[default]Path/IPAddressNat1" },
        "type": "tag"
      }
    },
    "custom.network.nat2": {
      "binding": {
        "config": { "mode": "indirect", "tagPath": "[default]Path/IPAddressNat2" },
        "type": "tag"
      }
    }
  },
  "root": {
    "meta": { "name": "root" },
    "type": "ia.container.coord"
  }
}
```

Config (`config.json`):

```json
{
  "UnusedCustomPropertiesRule": { "enabled": true, "kwargs": { "severity": "error" } }
}
```

Run:

```bash
poetry run python -m ignition_lint --config config.json --files "view.json"
```

## Actual behavior

```
❌ Found 1 errors:

  UnusedCustomPropertiesRule:
    • custom.network: custom property 'network' is defined but never referenced
```

## Expected behavior

`custom.network` should be considered **used**, because:
- Its child properties `custom.network.nat1` and `custom.network.nat2` have bindings (they populate `network`'s children), and
- Those children are referenced (`self.custom.network.nat1`, `self.custom.network.nat2`) in the `onChange` script.

A parent/container custom property should be treated as used whenever any of its descendant paths are bound or referenced. No error should be reported here.

## Suggested fix

When determining usage in `finalize`, treat a defined property as used if any tracked binding-owner path or reference path is a **child** of it (i.e. starts with `<prop_path>.`). Equivalently, when marking a reference/binding-owner as used, also mark each ancestor segment. This covers view-level (`view.custom.X`), view params (`view.params.X`), and component-level (`Component.custom.X`) object properties.

## Additional context

- Discovered on a real (proprietary) Perspective view where a structured object custom property holds nested IP-address fields bound via indirect tag bindings and consumed in an `onChange` script. Example above is reduced/sanitized with generic naming.
- Related: #84 ([FEATURE] Rule for `custom.*` prop settings).
